recipes/trantor/all/conanfile.py

```diff
@@ -52,59 +48,69 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("openssl/1.1.1q")
         if self.options.with_c_ares:
             self.requires("c-ares/1.18.1")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "14")
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
 
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
         if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("trantor requires C++14, which your compiler does not support.")
+            if Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
         else:
-            self.output.warn("trantor requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+            self.output.warn(f"{self.ref} requires C++{self._minimum_cpp_standard}. Your compiler is unknown. Assuming it supports C++{self._minimum_cpp_standard}.")
 
-        # TODO: Compilation succeeds, but execution of test_package fails on Visual Studio 16 MDd
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) == "16" and \
-           self.options.shared == True and self.settings.compiler.runtime == "MDd":
-            raise ConanInvalidConfiguration("trantor does not support the MDd runtime on Visual Studio 16.")
+        # TODO: Compilation succeeds, but execution of test_package fails on Visual Studio with MDd
+        if is_msvc(self) and self.options.shared and self.info.settings.compiler.runtime == "MDd":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support the MDd runtime on Visual Studio.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if Version(self.version) < "1.5.6":
+            tc.variables["BUILD_TRANTOR_SHARED"] = self.options.shared
+        else:
+            # Trantor's CMakeaLists.txt has BUILD_SHARED_LIBS option.
```

Comment:
Harmless typo in the comment here, `CMakeaLists.txt` -> `CMakeLists.txt`. Does `CMakeToolchain` not set `BUILD_SHARED_LIBS` automatically?

Reply:
@jwillikers 
Thank you for youre pointing out.
I will fix typo.

About BUILD_SHARED_LIB, I agree conan sets BUILD_SHARED_LIBS automatically.

But trantor/>=1.5.6 create option BUILD_SHARED_LIBS explicit.
https://github.com/an-tao/trantor/blob/master/CMakeLists.txt#L6

Without specifying BUILD_SHARED_LIB in conanfile.py, conan always create static library.
Please refer following:
https://c3i.jfrog.io/c3i/misc/logs/pr/13187/1-linux-gcc/trantor/1.5.7//8036752bc9b7fac6d6b8d40b2badcfb739586756-build.txt

Reply:
Aw, they must have set it as a standard variable instead of a cache variable, which would override any value set for a cache variable. Love it when that happens. Thanks!

Reply:
Good catch @toge ! 👏